File: Backend/Services/BloodBankService.cs
using BloodBankFinderApp.Models;
using BloodBankFinderApp.Repositories.Interfaces;
using BloodBankFinderApp.Services.Interfaces;

namespace BloodBankFinderApp.Services
{
    public class BloodBankService: IBloodBankService
    {
        IBloodBankRepo _bloodBankRepo;

        public BloodBankService(IBloodBankRepo bloodBankRepo)
        {
            _bloodBankRepo= bloodBankRepo;
        }


        public BloodBank CreateBloodBank(BloodBank bloodBank)
        {
            return  _bloodBankRepo.CreateBloodBank(bloodBank);
        }

        public bool DeleteBloodBank(string id)
        {
            return  _bloodBankRepo.DeleteBloodBank(id);
        }

        public BloodBank GetBloodBankById(string id)
        {
            return  _bloodBankRepo.GetBloodBankById(id);
        }

        public  List<BloodBank> GetBloodBanks()
        {
            return _bloodBankRepo.GetBloodBanks();
        }

        public bool UpdateBloodBank(string id, BloodBank bloodBankIn)
        {
            return  _bloodBankRepo.UpdateBloodBank(id, bloodBankIn);
        }
    }
}

File: FrontEnd/src/shared/services/bloodbank.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { bloodbank } from '../models/bloodbank';
import { BehaviorSubject, Observable } from 'rxjs';
import { urls } from 'src/urls';
import { BloodbankModule } from 'src/bloodbank/bloodbank.module';
@Injectable({
  providedIn: 'root'
})
export class BloodbankService {

  constructor(private http: HttpClient) { }

  getBloodBanks(): Observable<bloodbank[]> {
    return this.http.get<bloodbank[]>(`${urls.bloodbankUrl}`);
  }
  getbloodbankbyid(bloodbankid:string):Observable<bloodbank> {
    return this.http.get<bloodbank>(`${urls.bloodbankUrl}/${bloodbankid}`);
  }
  addBloodBank(bloodBank: bloodbank): Observable<bloodbank[]> {
    return this.http.post<bloodbank[]>(`${urls.bloodbankUrl}`, bloodBank);
  }

  deleteBloodBank(bloodBankId: string): Observable<bloodbank[]> {
    return this.http.delete<bloodbank[]>(`${urls.bloodbankUrl}/${bloodBankId}`);
  }

  updateBloodBank(bloodbankid:string,bloodBank: bloodbank): Observable<bloodbank> {
    return this.http.put<bloodbank>(`${urls.bloodbankUrl}/${bloodbankid}`, bloodBank,{headers: {'Content-Type':'application/json'}});
  }

 
}

